openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/OpenSearch.scala

```diff
@@ -21,75 +21,25 @@ import scala.collection.Map
 object OpenSearch {
   private val logger = LoggerFactory.getLogger(classOf[OpenSearch])
   private val requestCounter = new AtomicLong
-
-  def apply(endpoint: URL) = new OpenSearch(endpoint)
```

Comment:
This could also have been a factory method that tries to return the 'right' type of opensearch?